### PR TITLE
Add new addresses

### PR DESCRIPTION
**Pull request by anonmos1**
Please review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ jumps that could confuse logs, servers, Tor, i2p, etc.
 
 This package contains the sdwdate time fetcher and daemon. No
 installation on remote servers required. To avoid conflicts, this daemon
-should be not be enabled together with ntp or tlsdated.
```
Comment (adrelanos) [inline README.md]: Can you please update that in debian/control as well? (No matter if in the same or different pull request.)

README.md is actually generated from debian/control description.

Comment (anonmos1) [inline README.md]: done, new addresses are coming

**Comment by anonmos1** [comment]
what is the criteria for the pool 3? I could easily add like a hundred more stable addresses there

**Comment by anonmos1** [comment]
And I haven't checked whether the new ones work with curl --head or not. I guess you would do it anyway, let me know if you want me to check that too

**Comment by adrelanos** [comment]
Thank you for working on this!

Criteria generally is here:
https://www.whonix.org/blog/suggest-trustworthy-tor-hidden-services-as-time-sources-for-sdwdate

> What web servers do you consider trustworthy, to take great care of their visitors’ privacy, that are stable and that get great amounts of traffic, and most important, are reachable over .onion as a Tor Hidden Service?

trustworthy / reputation is of course a difficult topic which is up to interpretation. Perhaps add a few please and then let's see if we are on the same page?

**Comment by adrelanos** [comment]
Testing is welcome as well. There is:

* `/usr/share/sdwdate/unit_test`
* https://github.com/Whonix/sdwdate/blob/master/usr/share/sdwdate/unit_test
* Probably best to use the development version of sdwdate from git.

However, given the number of hidden services, that unit test probably needs some work. Not a great idea to request all onions at once and then wonder if some time out.

**Comment by anonmos1** [comment]
What about ones like this? http://fz474h2o46o2u7xj.onion/

I assume it has very little traffic since it is new but otherwise matches the criteria very well

**Comment by adrelanos** [comment]
Besides low traffic, should be okay. Good question. I am inviting comments on that one.

https://forums.whonix.org/t/suggest-trustworthy-tor-hidden-services-as-time-sources-for-sdwdate/856/33

**Comment by anonmos1** [comment]
Do you also test if these servers' times are more or less correct? Not x hour/day/week/year behind/ahead

**Comment by adrelanos** [comment]
Not really. But should.

In Whonix 14 sdwdate there there has a sanity test been implemented that makes sure times are within Tor consensus valid-after and valid-until. Otherwise another server is tried.

Indeed, better such servers don't even make it into the list. The unit test should be improved to test for these.

**Comment by anonmos1** [comment]
what about kkkkkkkkkk63ava6.onion and qubesos4rrrrz6n4.onion?

**Comment by anonmos1** [comment]
toristinkirir4xj.onion is on the new globaleaks list as well as pool 3, which one to keep?

**Comment by adrelanos** [comment]
> what about kkkkkkkkkk63ava6.onion and qubesos4rrrrz6n4.onion?

I am unsure about that. We already use them for apt-get updating. I wonder if that would open up for any anonymity attacks using the same source for time information as well as updates. (If we were to go by that argument, then we'd also have to remove Debian apt onion from the list.)

**Comment by adrelanos** [comment]
> toristinkirir4xj.onion is on the new globaleaks list as well as pool 3, which one to keep?

Should remain in pool 2 as globaleaks only then and not be duplicated in pool 3.

**Comment by anonmos1** [comment]
I was thinking of qdigse2yzvuglcix.onion but that's the default keyserver in Whonix

**Comment by anonmos1** [comment]
What is the benefit of having three pools instead of a single big pool and choosing 3 random addresses from it?

**Comment by anonmos1** [comment]
some suggestions, what do you think?

https://github.com/asciimoo/searx/wiki/Searx-instances
http://lelantoss7bcnwbv.onion/
http://torbox3uiot6wchz.onion/
http://secmailw453j7piv.onion/

**Comment by anonmos1** [comment]
I tested correctness of all of the servers' times. The incorrect ones are below:

pool 2

zjlzbo7y6nd2xbuz.onion	48 sec ahead
p6vbgbn7ggutkt3i.onion	22 min ahead
hgowugmgkiv2wxs5.onion	13 sec behind
wcnueib4qrsm544n.onion	32 sec ahead
w6csjytbrl273che.onion	2:17 min ahead
ppdz5djzpo3w5k2z.onion	4 min ahead
acabtd4btrxjjrvr.onion	50 sec ahead
5karyquenden4d6k.onion	4 min ahead
toristinkirir4xj.onion	6 hours ahead

pool 3

msydqstlz2kzerdg.onion	35 sec ahead
archivecrfip2lpi.onion	36 sec behind

**Comment by adrelanos** [comment]
> What is the benefit of having three pools instead of a single big pool and choosing 3 random addresses from it?

https://www.whonix.org/wiki/Sdwdate#sdwdate_source_pools

Essentially putting difference groups into distinct pools and thereby making it unlikely, that will collaborate. One compromised pool should not compromise everyone.

**Comment by adrelanos** [comment]
> I tested correctness of all of the servers' times.

If it is not too much to ask... Could you contact them to report the issue?

**Comment by adrelanos** [comment]
> https://github.com/asciimoo/searx/wiki/Searx-instances

Great!

> http://lelantoss7bcnwbv.onion/
> http://torbox3uiot6wchz.onion/
> http://secmailw453j7piv.onion/

Not so great on first look. These aren't reputable individuals or organizations. These are quite new and unpopular. It's good to know onion mail servers grow back again, but these could all be hosted by the same person and go down any time since these are anonymous hosts.

**Comment by adrelanos** [comment]
Please consider https://forums.whonix.org/t/suggest-trustworthy-tor-hidden-services-as-time-sources-for-sdwdate/856/36 i.e. please add heads, devuan, Whonix and Qubes onions.

**Comment by anonmos1** [comment]
secmail is new but lelantos and torbox are 4 years old and I think are good candidates

Should I temporarily remove all of the ones with incorrect times above? Do you consider like 13 or 32 seconds difference acceptable? I prefer if someone else contacts them with good explanation

Does sdwdate work with https? protonmail's address doesn't provide time over http, only https://protonirockerxow.onion

I think the mirrors from thomas white belongs to pool 3 along with torproject, I would add some other 6-7 mirrors from the same source
I would group whonix and qubes together, in pool 3
I would add qdigse2yzvuglcix.onion (which is default keyserver in torbirdy and whonix)
I would add searx instances to pool 3 separately

Tell me if you don't agree with any of them

Some suggestions
subgraphqov3womk.onion
scihub22266oqcxt.onion
grrmailb3fxpjbwm.onion
blockchainbdgpzk.onion?

parazite (If yes, which pool?)
kpynyvym6xqi7wz2.onion
qx7j2selmom4ioxf.onion (If yes, which pool?)

http://www.voidlinux.eu/news/2016/08/Tor-Hidden-Service.html
s7y2awkwau4nbdpu.onion
fd6dqrupy3af4xwb.onion

3kyl4i7bfdgwelmf.onion

jlve2y45zacpbz6s.onion

crypty22ijtotell.onion
cryjabkbdljzohnp.onion

wi7qkxyrdpu5cmvr.onion

clgs64523yi2bkhz.onion

privacyintyqcroe.onion

propub3r6espa33w.onion (together with its securedrop?)

**Comment by adrelanos** [comment]
If we are not sure about which pool, perhaps "the one that currently has
the least members" is the best answer.

anonmos1:
> Should I temporarily remove all of the ones with incorrect times
> above?

Yes, please.

> Do you consider like 13 or 32 seconds difference acceptable?

No.

> 
> Does sdwdate work with https?

No. Ticket:

https://phabricator.whonix.org/T133

> I think the mirrors from thomas white belongs to pool 3 along with
> torproject, I would add some other 6-7 mirrors from the same source I
> would group whonix and qubes together, in pool 3 I would add
> qdigse2yzvuglcix.onion (which is default keyserver in torbirdy and
> whonix) I would add searx instances to pool 3 separately

Seems sensible.

Looking into your other examples now.

**Comment by adrelanos** [comment]
> subgraphqov3womk.onion

Yes.

> scihub22266oqcxt.onion

Looks like a "warez" site. Probably best to stay away from any pages that obviously break any laws even if the law in question is of questionable value.

> grrmailb3fxpjbwm.onion

No content. No idea what that is.

> blockchainbdgpzk.onion?

If this is some blockchain domain that exists long term on clearnet as well, then yes.

Please make sure to add a comment for these ones where the onion domain was referenced. I.e. on the clearnet page or however. 

> parazite (If yes, which pool?)
> kpynyvym6xqi7wz2.onion
> qx7j2selmom4ioxf.onion (If yes, which pool?)

No warez.

> http://www.voidlinux.eu/news/2016/08/Tor-Hidden-Service.html

Yes.

> s7y2awkwau4nbdpu.onion

Yes. (As mirror of voidlinux.)

> fd6dqrupy3af4xwb.onion

Yes. (As mirror of voidlinux.)

> 3kyl4i7bfdgwelmf.onion

Probably yes. (If referenced on their clearnet page.)

> jlve2y45zacpbz6s.onion

Do they have a clearnet page as well?

> crypty22ijtotell.onion

Yes. (Cryptoparty group.)

> cryjabkbdljzohnp.onion

Yes. (Cryptoparty group.)

> wi7qkxyrdpu5cmvr.onion

Yes.

> clgs64523yi2bkhz.onion

Yes.

> privacyintyqcroe.onion

Yes. Has SSL, but that one should still work with sdwdate. See below.

> propub3r6espa33w.onion (together with its securedrop?)

Yes.

-----

```
curl --proto =http --head https://www.whonix.org
```
> `curl: (1) Protocol "https" not supported or disabled in libcurl`

```
curl --proto =http --head privacyintyqcroe.onion
```
> `Date: Wed, 08 Mar 2017 23:15:02 GMT`

**Comment by anonmos1** [comment]
try grrmailb3fxpjbwm.onion again. it is tor version of guerrillamail

jlve2y45zacpbz6s.onion is torstatus.rueckgr.at, at least 3+ years old

these?
lelantoss7bcnwbv.onion
torbox3uiot6wchz.onion

**Comment by anonmos1** [comment]
why did you post that curl command?

it works if you do this, is this not useful?

curl --head https://www.whonix.org

kkkkkkkkkk63ava6.onion and qubesos4rrrrz6n4.onion have incorrect times :)

**Comment by adrelanos** [comment]
I posted that curl command because... sdwdate does not speak ssl. By
using that command we force disable ssl. Make curl only use http. That
way curl can be tested to see if a ssl enabled onion will work with sdwdate.

> kkkkkkkkkk63ava6.onion and qubesos4rrrrz6n4.onion have incorrect times :)
> 

Will fix.

And it's the same server. Whonix hosts Qubes' onions. Therefore no need
to add both.

**Comment by anonmos1** [comment]
the list of torproject or debians addresses counts as one entry right?

I think it may help anonymity to have more onion addresses providing time source even if from the same server, at least shouldn't hurt?

**Comment by adrelanos** [comment]
anonmos1:
> the list of torproject or debians addresses counts as one entry
> right?

Yes.

> I think it may help anonymity to have more onion addresses providing
> time source even if from the same server, at least shouldn't hurt?

It would give that host a higher chance of being used, i.e. would give
them too much power in comparison to others without mirrors.

**Comment by anonmos1** [comment]
I was going to add it as a list so I guess there are no problems then?

**Comment by adrelanos** [comment]
No problems with a list.

**Comment by adrelanos** [comment]
> try grrmailb3fxpjbwm.onion again. it is tor version of guerrillamail

Yes.

> jlve2y45zacpbz6s.onion is torstatus.rueckgr.at, at least 3+ years old

Yes.

> these?
> lelantoss7bcnwbv.onion
> torbox3uiot6wchz.onion

No, since no one is accountable.

**Comment by anonmos1** [comment]
when checking some addresses with curl I get inconsistent results, like it shows me the time I requested the header, like 10-15-20 seconds ago but other times it gives me the exact time right now, this happened on a few addresses so far and I'm not sure if this is an issue with the address or the network or something

**Comment by adrelanos** [comment]
anonmos1:
> when checking some addresses with curl I get inconsistent results, like it shows me the time I requested the header, like 10-15-20 seconds ago but other times it gives me the exact time right now, this happened on a few addresses so far and I'm not sure if this is an issue with the address or the network or something
> 

Could be lag due to Tor connection speeds where nothing can be done
about it.

**Comment by anonmos1** [comment]
I think in that case it could be a good idea to avoid such addresses, I have seen only like 3-4 of them, I will check if it's temporary or not

what happened to whonix & qubes addresses? I can't reach them

**Comment by anonmos1** [comment]
ulrn6sryqaifefld.onion#https://searx.me https://github.com/asciimoo/searx/wiki/Searx-instances
eb6w5ctgodhchf3p.onion#https://s3arch.eu https://github.com/asciimoo/searx/wiki/Searx-instances

there are two new incorrect time sources

**Comment by adrelanos** [comment]
anonmos1:
> I think in that case it could be a good idea to avoid such addresses, I have seen only like 3-4 of them, I will check if it's temporary or not

Okay, great!

> what happened to whonix & qubes addresses? I can't reach them
> 

https://github.com/Whonix/Whonix/issues/412

**Comment by anonmos1** [comment]
4 seconds behind?

wlupld3ptjvsgwqw.onion

**Comment by adrelanos** [comment]
Yes.

**Comment by anonmos1** [comment]
should I add wlupld3ptjvsgwqw.onion or not?

**Comment by adrelanos** [comment]
> should I add wlupld3ptjvsgwqw.onion or not?

Yes.

> what is the max acceptable limit for time differences?

Good question! What's the average time drift that you have observed?

**Comment by anonmos1** [comment]
Right now pool 1 and pool 2 has 27 entries, pool 3 has 26 entries

I found some new addresses here: https://github.com/alecmuffett/onion-sites-that-dont-suck

**Comment by anonmos1** [comment]
>Good question! What's the average time drift that you have observed?
Apart from the ones I pointed out, I didn't observe any difference, perhaps 1 sec at most

**Comment by adrelanos** [comment]
> Right now pool 1 and pool 2 has 27 entries, pool 3 has 26 entries

That's a great balance!

(Theoretically speaking, 35 vs 25 vs 45 would also be fine. Just 35 vs 9 is a non-ideal state.)

> Apart from the ones I pointed out, I didn't observe any difference, perhaps 1 sec at most

Great! Answered here:
https://forums.whonix.org/t/suggest-trustworthy-tor-hidden-services-as-time-sources-for-sdwdate/856/40

**Comment by adrelanos** [comment]
> I found some new addresses here: https://github.com/alecmuffett/onion-sites-that-dont-suck

Answered here:
https://forums.whonix.org/t/suggest-trustworthy-tor-hidden-services-as-time-sources-for-sdwdate/856/41

**Comment by anonmos1** [comment]
I think it is ready to merge, please make final review and tests

**Comment by anonmos1** [comment]
what about these?
deepdot35wvmeyd5.onion
jxm5d6emw5rknovg.onion

**Comment by adrelanos** [comment]
(I'll reply to each of your comment one by one.)

anonmos1:
> what about these?
> deepdot35wvmeyd5.onion
> jxm5d6emw5rknovg.onion
> 

Please add.

**Comment by anonmos1** [comment]
yuxv6qujajqvmypv.onion?

**Comment by adrelanos** [comment]
anonmos1:
>  yuxv6qujajqvmypv.onion?
> 

Please add.

**Comment by anonmos1** [comment]
final balance
35-29-29

there are 22 more potential time sources if someone is willing to contact them with a good explanation

**Comment by anonmos1** [comment]
these are probably all I could find for now, tell me the ones you would like to see added

with good time:
http://qza32xuddl3guikc.onion/ https://thetinhat.com/
http://zerobinqmdqd236y.onion/ https://zerobin.net/
http://projpmcxufvim7be.onion/ http://wiki.project-pm.org/
http://74ypjqjwf6oejmax.onion/ https://beneathvt.com/
http://chloenlpvlemmmmd.onion/ https://chloe.re/
http://oxwugzccvk3dk6tj.onion/ https://8ch.net/
https://nmf6cg7tiyqlhsg3.onion/ http://rayservers.com/e-mail
http://mpf3i4k43xc2usxj.onion/ https://blog.samwhited.com/
http://theches3nacocgsc.onion/ https://thechess.ru/

with incorrect time (I will add as removed if you say "add"):
http://u75jkrt3umu2c7pn.onion/ https://poum.niij.org/
http://en.louhlbgyupgktsw7.onion/ http://onionmail.info https://github.com/onionmail/onionmail
http://torrentzwealmisr.onion/ https://torrentz2.eu/
http://x4torrentjjjjuxy.onion/ http://torrentproject.se/
http://y5fmhyqdr6r7ddws.onion/ https://anarplex.net/
http://vturtipc7vmz6xjy.onion/ https://ef.gy/
http://ponieslzi3ivbynd.onion/ http://derpy.fr/yp.html
http://shadow7jnzxjkvpz.onion/ http://shadowlife.cc/

**Comment by adrelanos** [comment]
anonmos1:
> these are probably all I could find for now, tell me the ones you would like to see added

I am amazed how much you dig up. :)

> with good time:
> http://qza32xuddl3guikc.onion/ https://thetinhat.com/

Yes.

> http://zerobinqmdqd236y.onion/ https://zerobin.net/
> http://projpmcxufvim7be.onion/ http://wiki.project-pm.org/
> http://74ypjqjwf6oejmax.onion/ https://beneathvt.com/
> http://chloenlpvlemmmmd.onion/ https://chloe.re/
> http://oxwugzccvk3dk6tj.onion/ https://8ch.net/
> https://nmf6cg7tiyqlhsg3.onion/ http://rayservers.com/e-mail
> http://mpf3i4k43xc2usxj.onion/ https://blog.samwhited.com/
> http://theches3nacocgsc.onion/ https://thechess.ru/

No.

> with incorrect time (I will add as removed if you say "add"):

That is a great idea.

> http://u75jkrt3umu2c7pn.onion/ https://poum.niij.org/

Yes.

> http://en.louhlbgyupgktsw7.onion/ http://onionmail.info https://github.com/onionmail/onionmail
> http://torrentzwealmisr.onion/ https://torrentz2.eu/
> http://x4torrentjjjjuxy.onion/ http://torrentproject.se/
> http://y5fmhyqdr6r7ddws.onion/ https://anarplex.net/
> http://vturtipc7vmz6xjy.onion/ https://ef.gy/
> http://ponieslzi3ivbynd.onion/ http://derpy.fr/yp.html
> http://shadow7jnzxjkvpz.onion/ http://shadowlife.cc/

No.

**Comment by anonmos1** [comment]
please test & merge

(when) will this be available to Whonix stable?

**Comment by adrelanos** [comment]
anonmos1:
> please test & merge

Let's see what comes out from the recent forum discussion first.

> (when) will this be available to Whonix stable?

At release of Whonix 14.

**Comment by anonmos1** [comment]
I added one active, one inactive address from that globaleaks list which had clearnet addresses, the rest only seem to exist among onion lists, I didn't touch onionmail

I think it is enough for now: 37-30-30

**Comment by anonmos1** [comment]
current state
pool 1: 37
pool 2: 36
pool 3: 35

**Comment by adrelanos** [comment]
This was merged.

**Pull request closed without merging**